com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/MaxRectsLayoutStrategy.java

```diff
@@ -92,17 +93,17 @@ private Page packPage(ArrayList<RectNode> inputRects) {
             minWidth = Math.min(minWidth, rect.getWidth());
             minHeight = Math.min(minHeight, rect.getHeight());
             if (settings.rotation) {
-                if ((rect.getWidth() > settings.maxPageWidth || rect.getHeight() > settings.maxPageHeight)
-                    && (rect.getWidth() > settings.maxPageHeight || rect.getHeight() > settings.maxPageWidth)) {
+                if ((rect.getWidth() > settings.maxPageWidth - settings.paddingX || rect.getHeight() > settings.maxPageHeight - settings.paddingY)
+                    && (rect.getWidth() > settings.maxPageHeight - settings.paddingY || rect.getHeight() > settings.maxPageWidth - settings.paddingX)) {
                     throw new RuntimeException("Image does not fit with max page size " + settings.maxPageWidth + "x" + settings.maxPageHeight
                         + " and padding " + settings.paddingX + "," + settings.paddingY + ": " + rect);
                 }
             } else {
-                if (rect.getWidth() > settings.maxPageWidth) {
+                if (rect.getWidth() > settings.maxPageWidth - settings.paddingX) {
                     throw new RuntimeException("Image does not fit with max page width " + settings.maxPageWidth + " and paddingX "
                         + settings.paddingX + ": " + rect);
                 }
-                if (rect.getHeight() > settings.maxPageHeight && (!settings.rotation || rect.getWidth() > settings.maxPageHeight)) {
```

Comment:
Second half of condition doesn't make any sense, because it's already in `!settings.rotation)` branch